controllers/yawol-controller/loadbalancer/loadbalancer_controller_test.go

```diff
@@ -388,6 +388,88 @@ var _ = Describe("loadbalancer controller", Serial, Ordered, func() {
 		})
 	}) // loadbalancerset context
 
+	When("set is ready", func() {
+		BeforeEach(func() {
+			timeNow := metav1.Now()
+			reconcileHash, err := helper.GetOpenStackReconcileHash(lb)
+			Expect(err).Should(Succeed())
+
+			lb.Status = yawolv1beta1.LoadBalancerStatus{
+				ExternalIP:             pointer.String("8.0.0.1"),
+				FloatingID:             pointer.String("floating-id"),
+				FloatingName:           pointer.String("floating-name"),
+				PortID:                 pointer.String("port-id"),
+				PortName:               pointer.String("port-name"),
+				SecurityGroupID:        pointer.String("sec-group-id"),
+				SecurityGroupName:      pointer.String("sec-group-name"),
+				LastOpenstackReconcile: &timeNow,
+				OpenstackReconcileHash: &reconcileHash,
+			}
+		})
+
+		It("should update the loadbalancer status", func() {
+			lbSet := yawolv1beta1.LoadBalancerSet{}
+			By("Wait for lbset")
+			Eventually(func(g Gomega) error {
+				lbSetList := &yawolv1beta1.LoadBalancerSetList{}
+				g.Expect(k8sClient.List(ctx, lbSetList, client.MatchingLabels(lb.Spec.Selector.MatchLabels))).Should(Succeed())
+				g.Expect(len(lbSetList.Items)).Should(Equal(1))
+				lbSet = lbSetList.Items[0]
+				return nil
+			}, timeout, interval).Should(Succeed())
+
+			By("Wait until replicas in lb object are 0")
+			var actual yawolv1beta1.LoadBalancer
+			Eventually(func() *int {
+				err := k8sClient.Get(ctx, lbNN, &actual)
+				if err != nil {
+					return nil
+				}
+				return actual.Status.Replicas
+			}, timeout, interval).Should(Equal(pointer.Int(0)))
+
+			Eventually(func() *int {
+				err := k8sClient.Get(ctx, lbNN, &actual)
+				if err != nil {
+					return nil
+				}
+				return actual.Status.ReadyReplicas
+			}, timeout, interval).Should(Equal(pointer.Int(0)))
+
+			By("Test - Patching status")
+			patch := client.MergeFrom(lbSet.DeepCopy())
+			lbSet.Status.Replicas = pointer.Int(1)
+			lbSet.Status.ReadyReplicas = pointer.Int(1)
+			Expect(k8sClient.Status().Patch(ctx, &lbSet, patch)).Should(Succeed())
+
+			By("Validate - ExternalIP should be set")
+			Eventually(func() *string {
+				err := k8sClient.Get(ctx, lbNN, &actual)
+				if err != nil {
+					return nil
+				}
+				return actual.Status.ExternalIP
+			}, timeout, interval).ShouldNot(BeNil())
```

Comment:
Why are we checking that `ExternalIP` is set in this test case? If I got it correctly, we set it ourselves earlier.

Reply:
👀 good question, it was done in the old tests of the status controller, but it makes no sense.